### ISSUE TITLE
API client cannot handle a list of countries passed to `get_litpop_default`

### ISSUE THREAD

**Issue by peanutfun** [CLIMADA-project/climada_python#485]
The API client method `get_litpop_default` accepts a string and a list of strings as `country` argument. However, passing a list with more than one element always results in an error of the form
```
climada.util.api_client.Client.AmbiguousResult: there are several datasets meeting the requirements
```

Here's a simple code example replicating this behavior:
```python
from climada.util.api_client import Client

client = Client()
litpop = client.get_litpop_default("CHE")  # Works
litpop = client.get_litpop_default(["CHE"])  # Works
litpop = client.get_litpop_default(["CHE", "AUT"])  # AmbiguousResult error
```

Error message (with linebreaks introduced for better readability):
```
Traceback (most recent call last):
  File "/Users/ldr.riedel/coding/climada_python/climada/test/test_api_client.py", line 202, in test_get_litpop
    litpop = client.get_litpop_default(country=["CHE", "AUT"], dump_dir=DATA_DIR)
  File "/Users/ldr.riedel/coding/climada_python/climada/util/api_client.py", line 845, in get_litpop_default
    return self.get_exposures(exposures_type='litpop', dump_dir=dump_dir, properties=properties)
  File "/Users/ldr.riedel/coding/climada_python/climada/util/api_client.py", line 777, in get_exposures
    dataset = self.get_dataset_info(data_type=exposures_type, name=name, version=version,
  File "/Users/ldr.riedel/coding/climada_python/climada/util/api_client.py", line 430, in get_dataset_info
    raise Client.AmbiguousResult("there are several datasets meeting the requirements:"
climada.util.api_client.Client.AmbiguousResult: there are several datasets meeting the requirements: 
[DatasetInfo(uuid='3681b54c-beb9-4737-b001-361d2071b4a4', data_type=DataTypeShortInfo(data_type='litpop', data_type_group='exposures'), name='LitPop_150arcsec_CHE', version='v1', status='active', properties={'res_arcsec': '150', 'exponents': '(1,1)', 'fin_mode': 'pc', 'spatial_coverage': 'country', 'date_creation': '2021-09-23', 'climada_version': 'v2.2.0', 'country_iso3alpha': 'CHE', 'country_name': 'Switzerland', 'country_iso3num': '756'}, files=[FileInfo(uuid='3681b54c-beb9-4737-b001-361d2071b4a4', url='https://data.iac.ethz.ch/climada/3681b54c-beb9-4737-b001-361d2071b4a4/LitPop_150arcsec_CHE.hdf5', file_name='LitPop_150arcsec_CHE.hdf5', file_format='hdf5', file_size=1280720, check_sum='md5:b22e66018a9590b48de597a8e7f9687b')], doi=None, description="LitPop asset value exposure per country: Gridded physical asset values by country, at a resolution of 150 arcsec. Values are total produced capital values disaggregated proportionally to the normalised product of nightlight intensity (Lit, based on NASA Earth at Night) and population count (Pop, based on Gridded Population of the World, Version 4.1). {'total_values': 'None', 'admin1_calc': 'False','reference_year': '2018', 'gpw_version': '4.11'}Reference: Eberenz et al., 2020. https://doi.org/10.5194/essd-12-817-2020", license='Attribution 4.0 International (CC BY 4.0)', activation_date='2021-09-13 09:09:33.493117+00:00', expiration_date=None),
DatasetInfo(uuid='98d4dfb2-a212-4fc0-ab5e-c62a5f7736ca', data_type=DataTypeShortInfo(data_type='litpop', data_type_group='exposures'), name='LitPop_150arcsec_AUT', version='v1', status='active', properties={'res_arcsec': '150', 'exponents': '(1,1)', 'fin_mode': 'pc', 'spatial_coverage': 'country', 'date_creation': '2021-09-23', 'climada_version': 'v2.2.0', 'country_iso3alpha': 'AUT', 'country_name': 'Austria', 'country_iso3num': '040'}, files=[FileInfo(uuid='98d4dfb2-a212-4fc0-ab5e-c62a5f7736ca', url='https://data.iac.ethz.ch/climada/98d4dfb2-a212-4fc0-ab5e-c62a5f7736ca/LitPop_150arcsec_AUT.hdf5', file_name='LitPop_150arcsec_AUT.hdf5', file_format='hdf5', file_size=1513248, check_sum='md5:f7ed66a24587a205c50adfee8b398ae8')], doi=None, description="LitPop asset value exposure per country: Gridded physical asset values by country, at a resolution of 150 arcsec. Values are total produced capital values disaggregated proportionally to the normalised product of nightlight intensity (Lit, based on NASA Earth at Night) and population count (Pop, based on Gridded Population of the World, Version 4.1). {'total_values': 'None', 'admin1_calc': 'False','reference_year': '2018', 'gpw_version': '4.11'}Reference: Eberenz et al., 2020. https://doi.org/10.5194/essd-12-817-2020", license='Attribution 4.0 International (CC BY 4.0)', activation_date='2021-09-13 09:09:41.882589+00:00', expiration_date=None)]
```

I expected `get_litpop_default` to behave exactly like `LitPop.from_countries`, where passing a list of countries as arguments results in a merged exposure of both countries.

**Comment by peanutfun** [comment]
@emanuel-schmid Do you want to fix this in the `get_dataset_info` method or should I simply implement the workaround using `Exposures.concat`?

**Comment by emanuel-schmid** [comment]
That even used to be possible! But at one point revised the `get_exposures` behavior and decided that it should always return an Exposures object corresponding to a  single climada dataset as provided by the API.

In the first version it returned a concatenated object from the combination of matching datasets. Back then `list_dataset_infos` was used to collect the datasets, not `get_dataset_info` as is used now, and `Exposures.concat` to combine them.
However, that led to inconsistent Exposures objects, in particular when properties other than 'country' were used to define a search query.
This was addressed and in a later version, it was still possible to concatenate Exposures, if certain properties of the listed datasets were uniform and others, like 'country', were divers. There was a configuration key for that: 'mutual_properties'.

But in the end this was given up as well, because it led to too many problems and was hard to maintain, especially while the datasets and their properties were very much a moving target. (They still are to some extent.)

However - since `get_litpop` is already a "special case" within the `api_client` methods, we may revive out-of-the-box-concatenation and see whether it's acceptable to allow multiple countries as argument for just this particular `Exposures` type.

**Comment by emanuel-schmid** [comment]
👍  Unless ...
What if we simply drop support for country lists? Who is going to be upset about it, given that a "real" list ended up in an error so far?

**Comment by peanutfun** [comment]
Fixed with #623
